### DISCUSSION TITLE
GitHub Roadmap review

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#218] [Maintainer discussions]
I moved our roadmap from the wiki to GitHub milestones and I'm looking for feedback on it.

[Here's the milestones page, sorted by due date](https://github.com/centerofci/mathesar/milestones?direction=asc&sort=due_date&state=open). This is the preferred view of the roadmap.

Some notes about how this is organized:
- Each milestone has a due date. We're shooting for the entire feature to be fully implemented by then (hopefully before then). The deadline for design to be completed is two weeks earlier than the milestone due date.
  - Due dates will be periodically readjusted to reflect reality.
- Milestone names contain an ordinal number so that we can easily see how they are ordered without having to look at the milestones page.
- If you don't know what milestone to put an issue in, but it should be done before the MVP ships, put it in the "Alpha Release" milestone.

Feedback that I'm looking for:
- Did we cover everything we need to do for the MVP?
- Do the due dates look reasonable for the information we have right now?
- Any general feedback about the format?

Tagging @ghislaineguerin @pavish @mathemancer @eito-fis for review.

**Comment by pavish** [comment]
Looks good to me.

> Did we cover everything we need to do for the MVP?

I currently don't see anything we might have missed.

> Do the due dates look reasonable for the information we have right now?

On the frontend aspect, it looks very reasonable.

We may cover certain milestones faster, and some slower depending on the amount of additional work required (eg., for the initial set of milestones, we'll need to create new components, which will take time but as we progress, we would get a lot faster). But the current milestone due dates seem to cover those additional time as well.

**Comment by eito-fis** [comment]
Seems good to me!

**Comment by ghislaineguerin** [comment]
Looks good!

**Comment by mathemancer** [comment]
I think the biggest strength of DabbleDB is the fact that it helps users iteratively and safely improve their data model, while being able to roll back and undo.  I don't see either of those covered in the roadmap.  Are we planning to put that after the MVP?  

I also think that some of the milestones (e.g., the join views) will have setting up data models as a dependency.  In case these are intended to be covered under milestone 7 -- working with tables, I think we'll need much more time for that, and it should probably be broken into smaller milestones.  Personally, I'd be willing to give up Kanban and Calendar visualizations in the MVP to get data modeling (at least some basics) in there.

**Reply by ghislaineguerin** [reply]
@mathemancer, I agree that improving the data model should be part of our vision. However, at the same time, I see features like Kanban or Calendar as good entry points for users who aren't aware of problems with their data models. I think of it as "trojan horse" features that bring people into Mathesar and expose them to the problems with their data and help them see how to solve them.

**Reply by kgodey** [reply]
@mathemancer I definitely meant to include both DDL (data modeling) and undo, I think they're essential to the MVP. I'll add milestones for them, thanks for flagging.

I don't think we should lose anything that's currently in the MVP plan.

**Reply by mathemancer** [reply]
That's great, I think we can do it (including the stuff I suggested taking out) by November if we're focused.

**Comment by kgodey** [comment]
I've updated the roadmap with Data Modeling and Undo & Redo milestones.
